### PR TITLE
Do not report CA1034 (Nested types should not be visible) for delegat…

### LINKED ISSUE #1347: CA1034 (Nested types should not be visible) reported for delegate declaration
#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

NestedTypesShouldNotBeVisibleAnalyzer

#### Repro steps

```
public class A
{
    public delegate void B();
}
```

#### Expected behavior
No CA1034

#### Actual behavior
`warning CA1034: Do not nest type B. Alternatively, change its accessibility so that it is not externally visible.
`

### PR DESCRIPTION
**Pull request by mavasani**
…e declarations

Fixes #1347

### CODE REVIEW

File: src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/NestedTypesShouldNotBeVisibleTests.cs [hunk 1]
```diff
@@ -361,6 +361,30 @@ End Class
...
+";
+            VerifyCSharp(code);
+        }
+
+        [Fact, WorkItem(1347, "https://github.com/dotnet/roslyn-analyzers/issues/1347")]
+        public void BasicNoDiagnosticDelegateTypeSpecialCases()
+        {
+            var code = @"
+Public Class A
+	Public Delegate Sub B()
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/NestedTypesShouldNotBeVisibleTests.cs]: Tabs -> spaces.

File: src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/NestedTypesShouldNotBeVisibleTests.cs [hunk 2]
```diff
@@ -361,6 +361,30 @@ End Class
...
+End Class
+";
+            VerifyBasic(code);
+        }
```
Comment (333fred) [inline src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/NestedTypesShouldNotBeVisibleTests.cs]: Whitespace on line below.

**Comment by mavasani** [comment]
@tmat `Pdb2Pdb.exe` is failing on all PRs on the repo. Is this known?

**Comment by tmat** [comment]
@mavasani Yes, sorry about that. Forgot to push a fix on Fri: https://github.com/dotnet/roslyn-analyzers/pull/1351

**Comment by mavasani** [comment]
test this please

**Pull request merged**